Apps/Src/Voltage.c

```diff
@@ -82,10 +113,21 @@ void Voltage_UpdateMeasurements(void){
   	assertOSError(err);
 	LTC6811_rdcv_safe(0, NUM_MINIONS, Minions); // Set to read back all cell voltage registers
 	//copies values from cells.c_codes to private array
+
+	// run median filter
+	for(int i = 0; i < NUM_BATTERY_MODULES; i++){
+		rawVoltages[i][voltFilterIdx] = Minions[i / MAX_VOLT_SENSORS_PER_MINION_BOARD].cells.c_codes[i % MAX_VOLT_SENSORS_PER_MINION_BOARD];
+	}
+	voltFilterIdx = (voltFilterIdx + 1) % VOLT_MEDIAN_FILTER_DEPTH;
+
+	// update public voltage values
 	OSMutexPend(&Voltage_Mutex, 0, OS_OPT_PEND_BLOCKING, &ts, &err);
   	assertOSError(err);
 	for(int i = 0; i < NUM_BATTERY_MODULES; i++){
-		VoltageVal[i] = Minions[i / MAX_VOLT_SENSORS_PER_MINION_BOARD].cells.c_codes[i % MAX_VOLT_SENSORS_PER_MINION_BOARD];
+		// you need to change this if you change VOLT_MEDIAN_FILTER_DEPTH
```

Comment:
I don't think it makes sense to have filter depth change since the actual filter function can't handle different values. Maybe remove these comments and just have a comment at the #define? If it could handle multiple values then we should consider a filter.h file.

Reply:
I left this comment, so it will be easier to see if that this will break if we change the `#define`. There is also a comment at the `#define`. I agree that we should create a separate file for filters if we start to mess with them a lot.